x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/components/agent_upgrade_status.tsx

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useMemo } from 'react';
+// import styled from 'styled-components';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { EuiBadge, EuiFlexGroup, EuiFlexItem, EuiIconTip } from '@elastic/eui';
+
+import type { AgentUpgradeDetails } from '../../../../../../../common/types';
+
+function getUpgradeStartDelay(scheduledAt?: string) {
+  return Math.round((Date.parse(scheduledAt || '') - Date.now()) / 36e5);
```

Comment:
a comment would be nice what this function returns (upgrade start delay in hours?)